GitUI/GitUICommands.cs

```diff
@@ -727,23 +727,30 @@ private bool StartResetChangesDialog(string fileName)
                 // Also delete new files, if requested.
                 if (resetAction == FormResetChanges.ActionEnum.ResetAndDelete)
                 {
-                    try
+                    string errorCaption = null;
+                    string errorMessage = null;
+                    string path = _fullPathResolver.Resolve(fileName);
+                    if (File.Exists(path))
                     {
-                        string path = _fullPathResolver.Resolve(fileName);
-                        if (File.Exists(path))
+                        try
                         {
                             File.Delete(path);
                         }
-                        else
+                        catch (Exception ex)
                         {
-                            Directory.Delete(path, true);
+                            errorCaption = Strings.ErrorCaptionFailedDeleteFile;
+                            errorMessage = ex.Message;
                         }
                     }
-                    catch (IOException)
+                    else
                     {
+                        errorCaption = Strings.ErrorCaptionFailedDeleteFolder;
+                        path.TryDeleteDirectory(out errorMessage);
                     }
-                    catch (UnauthorizedAccessException)
+
+                    if (errorMessage is { })
                     {
+                        MessageBox.Show(null, errorMessage, errorCaption, MessageBoxButtons.OK, MessageBoxIcon.Error);
```

Comment:
🤔 ❗️  This may be _extremely_ invasive if attempt to reset a lot of files/folders that aren't accessible for whatever reason...

Reply:
Maybe not... I haven't followed the code all the way through. 😴